### PR TITLE
MINOR: Conditional build for block with change

### PR DESCRIPTION
**Pull request by bbejeck**
### Description
This PR adds [modularized builds](https://app.asana.com/0/0/1203752202076449/f) to Kafka Tutorials, meaning that only the pipeline block with changes will run during the build. 

There are cases where the complete build will run:

1. Any change to the `semaphore.yml` file. Note that the test PRs contained `{pipeline_file: 'ignore'}` as the default behavior for semaphore is to run the entire pipeline when there are changes to the `semaphore.yml` file.
2. Merging a PR into the `master` branch.
3. Merging the `master` branch into the `release` branch.

### Testing
For testing, I pushed several dummy PRs matching different scenarios:

1. [Kafka Streams / Kafka only changes](https://confluentinc.semaphoreci.com/workflows/b388d759-8608-42fb-9de9-6dae776734ea?pipeline_id=91a88593-cfdf-41c8-84c1-da152754d194)
2. [ksqlDB changes](https://confluentinc.semaphoreci.com/workflows/53a0a401-259a-4511-a0a9-acfd5be4f529?pipeline_id=0003257f-4523-462c-aec7-077f4c2d038b)
3. [ksqlDB recipe changes](https://confluentinc.semaphoreci.com/workflows/4a1ebe7c-a1e6-48ca-8330-c213fd744cf9?pipeline_id=3a4d46ad-69ed-49c3-8887-631273ab6e9e)
4. [Kafka Streams and ksqlDB changes](https://confluentinc.semaphoreci.com/workflows/5fefd5c6-706f-4e37-a101-e541a0025461?pipeline_id=6287a2ff-b81c-437a-9d4a-78808856e4bb)
5. [Kafka Streams data-harness file changes](https://confluentinc.semaphoreci.com/workflows/50a6ed2a-efbe-46f3-96c6-85fac408247a?pipeline_id=41b50928-cd3a-443b-a227-7bd7dd42240f)
6. [Kafka Streams and ksqlDB recipe data-harness changes](https://confluentinc.semaphoreci.com/workflows/ab3df420-e6e2-4d55-a3a6-8cb93b41d3ac?pipeline_id=c7c9e4e8-2446-4c20-b168-ed990f1fb414)
7. [Changes to all harness files](https://confluentinc.semaphoreci.com/workflows/545f01ba-7a56-4534-ab9b-92a1b450f6fc?pipeline_id=c81b746a-72b1-47a6-8729-22d04913e1a4)

### CODE REVIEW

File: .semaphore/semaphore.yml [hunk 1]
```diff
@@ -115,8 +117,10 @@ blocks:
 
   - name: Run first block of tests (mostly Kafka and KStreams)
     dependencies: ["☕️ Build and Test Java (Kafka and Kafka Streams) Only"]
+    run:
+      when: "(branch = 'master' OR branch = 'release') OR change_in(['../_includes/tutorials/**/kstreams/**/*', '../_includes/tutorials/**/kafka/**/*', '../_includes/tutorials/**/scala/**/*', '../_data/harnesses/**/kstreams.yml', '../_data/harnesses/**/kafka.yml'])"
     execution_time_limit:
-      minutes: 20
+      minutes: 10
```
Comment (bbejeck) [inline .semaphore/semaphore.yml:123]: Reducing the timeout for builds to 10 minutes should be long enough.  We still have flaky tests that won't pass hence holding up the build artificially.

**Comment by bbejeck** [comment]
Single test failure unrelated

File: .semaphore/semaphore.yml [hunk 2]
```diff
@@ -99,6 +99,8 @@ blocks:
 
   - name: "☕️ Build and Test Java (Kafka and Kafka Streams) Only"
     dependencies: ["Build the website"]
+    run:
+      when: " (branch = 'master' OR branch = 'release') OR change_in(['../_includes/tutorials/**/kstreams/**/*', '../_includes/tutorials/**/kafka/**/*', '../_includes/tutorials/**/scala/**/*'])"
```
Comment (davetroiano) [inline .semaphore/semaphore.yml:103]: nittiest of nits: remove space after first double quote

**Review by davetroiano — APPROVED** [review]
LGTM thanks for taking this one and testing out various scenarios

**Comment by bbejeck** [comment]
Failure unrelated

**Pull request merged**